### ISSUE TITLE
Add support for matrix / parameterization

### ISSUE THREAD

**Issue by theseion** [coreruleset/go-ftw#321]
Many of the tests are duplicates with minor variations in the payload. Test for many rules could be condensed to two or three tests with many payloads instead.

- extend the tests schema with support for a list / matrix of payloads
- add support in go-ftw

**Comment by fzipi** [comment]
Yeah, I wanted some time ago to use the same concept as nuclei-templates, with golang templating.

So adding a templated test, with `data: {{ payload }}` and a way to have tests generated for each payload...
